src/exportJarSteps/ExportJarTaskProvider.ts

```diff
@@ -57,13 +135,173 @@ class ExportJarTaskTerminal implements Pseudoterminal {
     }
 
     public async open(initialDimensions: TerminalDimensions | undefined): Promise<void> {
+        if (_.isEmpty(this.stepMetadata.outputPath)) {
+            // Only for custom task
+            this.stepMetadata.outputPath = (Settings.getExportJarTargetPath() === ExportJarProperties.SETTING_ASKUSER) ?
+                ExportJarProperties.SETTING_ASKUSER : join(this.stepMetadata.workspaceFolder.uri.fsPath, this.stepMetadata.workspaceFolder.name + ".jar");
+        }
+        if (!_.isEmpty(this.stepMetadata.elements)) {
+            const classPathMap: Map<string, string[]> = new Map<string, string[]>();
+            const testClassPathMap: Map<string, string[]> = new Map<string, string[]>();
+            const runtimeDependencies: string[] = [];
+            const testDependencies: string[] = [];
+            const projectList: INodeData[] = await Jdtls.getProjects(this.stepMetadata.workspaceFolder.uri.toString());
+            for (const project of projectList) {
+                const runtimeClassPaths: string[] = [];
+                await this.getClasspaths(project.uri, "runtime", runtimeClassPaths, runtimeDependencies);
+                classPathMap.set(project.name, runtimeClassPaths);
+                const testClassPaths: string[] = [];
+                await this.getClasspaths(project.uri, "test", testClassPaths, testDependencies);
+                testClassPathMap.set(project.name, testClassPaths);
+            }
+            this.stepMetadata.classpaths = await this.resolveClassPaths(runtimeDependencies, testDependencies, classPathMap, testClassPathMap);
+        }
         await createJarFile(this.stepMetadata);
         this.closeEmitter.fire();
     }
 
     public close(): void {
 
     }
+
+    private toAbsolute(path: string): string {
+        if (path.length > 0 && path[0] === "!") {
```

Comment:
1. find the first index of `!`
2. if the index >= 0, set a boolean flag to true.
3. if the index >= 1, log a warning to tell the user that multiple `!` are not allowed and will be ignored.
4. extract a substring after the index
5. join if is not absolute
6. Add `!` if the boolean flag is true